src/bika/lims/browser/header_table.py

```diff
@@ -61,28 +61,65 @@ def __call__(self):
             schema = self.context.Schema()
             fields = schema.fields()
             form = self.request.form
+
+            errors = {}
+            field_values = {}
             for field in fields:
-                fieldname = field.getName()
-                if fieldname in form:
-                    # Handle (multiValued) reference fields
-                    # https://github.com/bikalims/bika.lims/issues/2270
-                    uid_fieldname = "{}_uid".format(fieldname)
-                    if uid_fieldname in form:
-                        value = form[uid_fieldname]
-                        if field.multiValued:
-                            value = value.split(",")
-                        field.getMutator(self.context)(value)
-                    else:
-                        # other fields
-                        field.getMutator(self.context)(form[fieldname])
-            message = _("Changes saved.")
-            # reindex the object after save to update all catalog metadata
-            self.context.reindexObject()
-            # notify object edited event
-            event.notify(ObjectEditedEvent(self.context))
-            self.context.plone_utils.addPortalMessage(message, "info")
+                value = self.get_field_value(field, form)
+                if value is None:
+                    continue
+
+                # Cache the field value in the request
+                self.request[field.getName()] = value
+
+                # Validate the field values
+                err = field.validate(value, self.context)
+                if err:
+                    # Clear field values and add the error
+                    errors.update({field.getName(): err})
+                    field_values = {}
+
+                elif not errors:
```

Comment:
shouldn't that be a single `else` ?

Reply:
If it was `else`, valid values would be stored on submit, while those not valid would not. I thought is better to not store anything unless all fields are valid

Reply:
Umm, you're right, is meaningless beacuse we already check later if there are errors